### PR TITLE
Add tolerance on www websites

### LINKED ISSUE #18: Don't display websites with `www`
Currently, when we go into the preferences, we can see a long list of websites to track and a lot are duplicated versions with and without `www`.

We can think of removing websites starting with `www` and track websites with a tolerance on this subdomain.

This will allow to remove duplicates in the list of websites

### PR DESCRIPTION
**Pull request by KillianKemps**
Fix #18 

Allow to track www websites even when the `www.` subdomain is not specified.

It allows to remove duplicates in the websites list.

### CODE REVIEW

**Comment by elegaanz** [comment]
Nice PR! I just found two small bugs:

- when you click the "Start tracking" button in the popup, we create fake data (1 second on the current website), but we don't ignore the `www` here, so we can have `www.example.org` which will always be `1` and `example.org` which will be the actual time.
- The "Start tracking" button shows on some website where it shouldn't. For instance, if you go on `www.youtube.com`, you will see it, because `www` is not ignored when checking if the site is tracked.

**Comment by KillianKemps** [comment]
Oh, good catch, I didn't checked this part :D
I will also work on the input to always convert websites to simple hostnames without `www`

**Comment by elegaanz** [comment]
I found two other little problems. First, we can add a website with `www` in the options.

The other problem is that when you open the popup on `www` website (like `www.youtube.com`), you'll first see the message telling you there is no data, because `hostname`/`activeUrl` is still with the `www`.

I think we could have a function that removes the `www` if needed, that we use it on every URL we have.

**Comment by KillianKemps** [comment]
I just fixed the possibility to add a website with `www` in the options :D.

Yes, I will change that too for the type of case like `www.youtube.com`.

Just a question, I had to fix that https://github.com/Qowala/browser-extension/pull/19/commits/1e56c9cec8565b30060b248fcb10f614c65b3249 that you added in your last PR. Why did you changed that to https://github.com/Qowala/browser-extension/pull/20/commits/aca0c723e4e6343c4b50c246654875758cbe923c#diff-763aa8deb604d725a74f8304903c1422R11 ?

File: lib/options.js [hunk 1]
```diff
@@ -8,13 +8,20 @@ const app = new Vue({
   },
   computed: {
     error: function () {
-      return this.websiteInput.length === 0 || !this.websiteInput.match(urlRegex)
+      return this.websiteInput.length && !this.websiteInput.match(urlRegex)
+    },
+    hostname: function () {
+      var hostname = new URL(`http://${this.websiteInput}`).hostname
```
Comment (elegaanz) [inline lib/options.js]: Use `let` please

**Comment by elegaanz** [comment]
> Just a question, I had to fix that 1e56c9c that you added in your last PR. Why did you changed that to aca0c72#diff-763aa8deb604d725a74f8304903c1422R11 ?

I think I did an error while writing the tests, so they failed, but I thought it was because of the function, not of the tests, so I modified it to make it work. :grin:

**Comment by KillianKemps** [comment]
So, @Bat41 I think it should be good now :D 
Do you think I should also add something which removes all `www.` websites for the users who already have the plugin installed? Or should I let them delete them themselves if they wish?

**Comment by elegaanz** [comment]
Everything seems to work fine! I think it is better to remove the `www.` websites automatically, and remove this feature in a few versions.

I let you merge.

**Comment by KillianKemps** [comment]
Hop, I added a function to clean the hostnames in the background script so I'm sure it will run at least once on all our users.

**Pull request merged**